ytypes/node.go

```diff
@@ -195,18 +194,35 @@ func retrieveNodeContainer(schema *yang.Entry, root interface{}, path *gpb.Path,
 			return retrieveNode(cschema, fv.Interface(), util.TrimGNMIPathPrefix(path, p[0:to]), np, args)
 		}
 
-		for _, p := range schPaths {
-			if !util.PathMatchesPrefix(path, p) {
-				continue
+		args := args
+		if args.reverseShadowPath {
+			schPaths := util.ShadowSchemaPaths(ft)
```

Comment:
I found this logic a bit hard to follow. 

iCan you explain why on line 221 when we have `!args.reverseShadowSchemaPaths` we check the shadow paths?

Reply:
I added some comments, hopefully it clarifies the logic.